dask_xgboost/core.py

```diff
@@ -298,6 +316,35 @@ def fit(self, X, y=None, classes=None):
             Feature Matrix. May be a dask.array or dask.dataframe
         y : array-like
             Labels
+        eval_set : list, optional
+            A list of (X, y) tuple pairs to use as validation sets, for which
```

Comment:
The code above (line 90) to pack them into a `xgb.DMatrix` currently assumes they are `dask.arrays`.   I remember your comment in the issue about this potentially being problematic, but I got it working with the unit test below.  Should we support NumPy arrays here as well?

Reply:
This makes things more complex.  Are they going to be large?  If so, do we need to worry about splitting them up in some way?  If not, then do we want to support dask inputs here?

Reply:
They will be large since they need to match the training data. Somewhere on the order of (100_000_000, 3), where the columns shouldn't be more than 3  (target variable and any weights).

Reply:
Can I do something similar to [this](https://github.com/dask/dask-xgboost/blob/master/dask_xgboost/core.py#L145) to split them up?

Reply:
Do we need to align the chunks of this array with the chunks of the training data?  Is this data supposed to be co-aligned?

Reply:
I just asked some folks here who are asking for this.  They are separate datasets entirely, so they can be distributed differently.  We shouldn't need to align the chunks to the training data.

Reply:
`100_000_000, 3` is also small enough that we might keep it in memory. 

I took the extra five seconds to actually read the definition of what this is.  My sense is that you're going to want validation data to be uniform across all of your workers.  I suggest that we just keep this as a numpy array and not deal with splitting it up.

Reply:
Good point.  Maybe we just go with numpy for now and see how it goes?

Reply:
I think that that is probably a good way to start.

Reply:
> I suggest that we just keep this as a numpy array and not deal with splitting it up.

Sorry, my head is spread thin this morning.  You already said that.  I'll make the changes.  Thank you for looking at this with me.